javax.validation分组能力支持问题

1、插件版本：2.5.1.191.0，IDEA版本：2021.3.3
2、移除勾选javax.validation, 勾选javax.validation(grouped)
3、代码定义
入参：
```
@Data
public class UserDTO {

    /**
     * 用户名
     */
    @NotBlank(groups = {GetValidator.class})
    private String username;

    /**
     * 密码
     */
    private String password;
}
```
GetValidator:
```
public interface GetValidator extends Default {
}
```
方法：
```
    /**
     * 添加
     * @param user 参数   {@link UserDTO}
     * @return 返回结果 {@link R<String>>}
     */
    @GetMapping("/get")
    public R<String> addUser(@Validated(value = GetValidator.class) UserDTO user) {
        return R.SUCCESS("删除成功！");
    }
```
3、上传接口报错
```
[ERROR]	error eval script:
def annMaps = it.annMaps("javax.validation.constraints.NotBlank")
groovy:
for(annMap in annMaps){
def fieldGroups = annMap["groups"] ?: [helper.findClass("javax.validation.groups.Default")]
def paramGroups = session.get("json-group") ?: [helper.findClass("javax.validation.groups.Default")]
for(fieldGroup in fieldGroups){
for(paramGroup in paramGroups){
if(fieldGroup.isExtend(paramGroup.name())){
return true
}
}
}
}
return false
[TRACE]	javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:158)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.eval(ScriptRuleParser.kt:99)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.access$eval(ScriptRuleParser.kt:36)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:67)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:34)
	at com.itangcent.intellij.config.rule.BooleanRule$filterWith$1.invoke(BooleanRule.kt:17)
	at com.itangcent.intellij.config.rule.BooleanRule$filterWith$1.invoke(BooleanRule.kt:5)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:34)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$RuleChainImpl.compute(DefaultRuleComputer.kt:74)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY$compute$1.apply(RuleMode.kt:55)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY$compute$1.apply(RuleMode.kt:51)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY.compute(RuleMode.kt:56)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.doComputer(DefaultRuleComputer.kt:55)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:31)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:10)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:34)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.access$computerListener(RuleComputeListenerRegistry.kt:11)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:31)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:11)
	at com.itangcent.idea.utils.ContextualPsiClassHelper$InnerComputeListener.computer(ContextualPsiClassHelper.kt:246)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:30)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computer(RuleComputeListenerRegistry.kt:24)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:29)
	at com.itangcent.intellij.config.rule.RuleComputer$DefaultImpls.computer(RuleComputer.kt:16)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:10)
	at com.itangcent.intellij.config.rule.RuleComputerKt.computer(RuleComputer.kt:27)
	at com.itangcent.idea.utils.CustomizedPsiClassHelper.afterParseFieldOrMethod(CustomizedPsiClassHelper.kt:45)
	at com.itangcent.idea.plugin.api.export.yapi.YapiPsiClassHelper.afterParseFieldOrMethod(YapiPsiClassHelper.kt:84)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$doGetFields$2.invoke(AbstractPsiClassHelper.kt:589)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$doGetFields$2.invoke(AbstractPsiClassHelper.kt:31)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.foreachField(AbstractPsiClassHelper.kt:628)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetFields(AbstractPsiClassHelper.kt:569)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetTypeObject(AbstractPsiClassHelper.kt:525)
	at com.itangcent.intellij.psi.DefaultPsiClassHelper.doGetTypeObject(DefaultPsiClassHelper.kt:234)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetTypeObject(AbstractPsiClassHelper.kt:286)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.getTypeObject(AbstractPsiClassHelper.kt:138)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$raw$1.invoke(RequestClassExporter.kt:849)
	at com.itangcent.common.utils.ExtensibleKt.cache(Extensible.kt:97)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.raw(RequestClassExporter.kt:847)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.processMethodParameters(RequestClassExporter.kt:503)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.exportMethodApi(RequestClassExporter.kt:195)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.access$exportMethodApi(RequestClassExporter.kt:46)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:150)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:46)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:151)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:32)
	at com.itangcent.intellij.context.ActionContext$callInReadUI$1$1.invoke(ActionContext.kt:462)
	at com.itangcent.common.concurrent.ValueHolder.compute(Holder.kt:100)
	at com.itangcent.intellij.context.ActionContext$callInReadUI$1.run(ActionContext.kt:460)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:865)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.itangcent.intellij.context.ActionContext.callInReadUI(ActionContext.kt:453)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:150)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:32)
	at com.itangcent.intellij.context.ActionContext$runAsync$2.run(ActionContext.kt:221)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:320)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
	... 74 more
Caused by: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:46)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at Script412.run(Script412.groovy:9)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
	... 75 more

```

单个分组可以成功，但是若是一个字段有多个分组如何配置？
```
@NotBlank(groups = {GetValidator.class, AddValidator.class})
```
这种依然报错？
```
run easy-yapi
[INFO]	Start find apis...
[TRACE]	No config be found
[DEBUG]	use recommend config
[INFO]	search api from: com.sitech.bds.portal.controller.GebVisitController
[TRACE]	No config be found
[DEBUG]	use recommend config
[TRACE]	No config be found
[DEBUG]	use recommend config
[INFO]	search api from: com.sitech.bds.portal.controller.GebVisitController
[ERROR]	error eval script:
def annMaps = it.annMaps("javax.validation.constraints.NotBlank")
groovy:
for(annMap in annMaps){
def fieldGroups = annMap["groups"] ?: [helper.findClass("javax.validation.groups.Default")]
def paramGroups = session.get("json-group") ?: [helper.findClass("javax.validation.groups.Default")]
for(fieldGroup in fieldGroups){
for(paramGroup in paramGroups){
if(fieldGroup.isExtend(paramGroup.name())){
return true
}
}
}
}
return false
[TRACE]	javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:158)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.eval(ScriptRuleParser.kt:99)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.access$eval(ScriptRuleParser.kt:36)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:67)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:34)
	at com.itangcent.intellij.config.rule.BooleanRule$filterWith$1.invoke(BooleanRule.kt:17)
	at com.itangcent.intellij.config.rule.BooleanRule$filterWith$1.invoke(BooleanRule.kt:5)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:36)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:34)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$RuleChainImpl.compute(DefaultRuleComputer.kt:74)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY$compute$1.apply(RuleMode.kt:55)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY$compute$1.apply(RuleMode.kt:51)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.itangcent.intellij.config.rule.BooleanRuleMode$ANY.compute(RuleMode.kt:56)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.doComputer(DefaultRuleComputer.kt:55)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:31)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$computer$1.invoke(DefaultRuleComputer.kt:10)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:34)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.access$computerListener(RuleComputeListenerRegistry.kt:11)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:31)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry$computerListener$1.invoke(RuleComputeListenerRegistry.kt:11)
	at com.itangcent.idea.utils.ContextualPsiClassHelper$InnerComputeListener.computer(ContextualPsiClassHelper.kt:246)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computerListener(RuleComputeListenerRegistry.kt:30)
	at com.itangcent.idea.utils.RuleComputeListenerRegistry.computer(RuleComputeListenerRegistry.kt:24)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:29)
	at com.itangcent.intellij.config.rule.RuleComputer$DefaultImpls.computer(RuleComputer.kt:16)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:10)
	at com.itangcent.intellij.config.rule.RuleComputerKt.computer(RuleComputer.kt:27)
	at com.itangcent.idea.utils.CustomizedPsiClassHelper.afterParseFieldOrMethod(CustomizedPsiClassHelper.kt:45)
	at com.itangcent.idea.plugin.api.export.yapi.YapiPsiClassHelper.afterParseFieldOrMethod(YapiPsiClassHelper.kt:84)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$doGetFields$2.invoke(AbstractPsiClassHelper.kt:589)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$doGetFields$2.invoke(AbstractPsiClassHelper.kt:31)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.foreachField(AbstractPsiClassHelper.kt:628)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetFields(AbstractPsiClassHelper.kt:569)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetTypeObject(AbstractPsiClassHelper.kt:525)
	at com.itangcent.intellij.psi.DefaultPsiClassHelper.doGetTypeObject(DefaultPsiClassHelper.kt:234)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.doGetTypeObject(AbstractPsiClassHelper.kt:286)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.getTypeObject(AbstractPsiClassHelper.kt:138)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$raw$1.invoke(RequestClassExporter.kt:849)
	at com.itangcent.common.utils.ExtensibleKt.cache(Extensible.kt:97)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.raw(RequestClassExporter.kt:847)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.processMethodParameters(RequestClassExporter.kt:503)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.exportMethodApi(RequestClassExporter.kt:195)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter.access$exportMethodApi(RequestClassExporter.kt:46)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:150)
	at com.itangcent.idea.plugin.api.export.core.RequestClassExporter$doExport$1$1.invoke(RequestClassExporter.kt:46)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:151)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1$1.invoke(ClassApiExporterHelper.kt:32)
	at com.itangcent.intellij.context.ActionContext$callInReadUI$1$1.invoke(ActionContext.kt:462)
	at com.itangcent.common.concurrent.ValueHolder.compute(Holder.kt:100)
	at com.itangcent.intellij.context.ActionContext$callInReadUI$1.run(ActionContext.kt:460)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:865)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.itangcent.intellij.context.ActionContext.callInReadUI(ActionContext.kt:453)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:150)
	at com.itangcent.idea.plugin.api.ClassApiExporterHelper$foreachMethod$1$1.invoke(ClassApiExporterHelper.kt:32)
	at com.itangcent.intellij.context.ActionContext$runAsync$2.run(ActionContext.kt:221)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:320)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
	... 74 more
Caused by: groovy.lang.MissingMethodException: No signature of method: com.intellij.psi.impl.source.PsiClassImpl.isExtend() is applicable for argument types: (String) values: [com.sitech.bds.portal.entity.GetValidator]
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:46)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at Script139.run(Script139.groovy:9)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
	... 75 more

[WARN]	error to resolve class:String>
```

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
@@ -131,7 +131,8 @@ abstract class ScriptRuleParser : RuleParser {
     }
 
     /**
-     * support is js:
+     * support usages:
+     *
      * it.name():String
      * it.hasAnn("annotation_name"):Boolean
      * it.ann("annotation_name"):String?
@@ -209,6 +210,14 @@ abstract class ScriptRuleParser : RuleParser {
             }
 
             if (obj is Collection<*>) {
+                val copy = LinkedList<Any?>()
+                for (ele in obj) {
+                    copy.add(wrap(ele))
+                }
+                return copy
+            }
+
+            if (obj is Array<*>) {
                 val copy = LinkedList<Any?>()
                 for (ele in obj) {
                     copy.add(wrap(ele))
@@ -349,7 +358,7 @@ abstract class ScriptRuleParser : RuleParser {
             return "class"
         }
 
-        open fun methods(): Array<ScriptPsiMethodContext> {
+        override fun methods(): Array<ScriptPsiMethodContext> {
             return jvmClassHelper!!.getAllMethods(psiClass)
                 .mapToTypedArray { ScriptPsiMethodContext(it) }
         }
@@ -358,7 +367,7 @@ abstract class ScriptRuleParser : RuleParser {
             return jvmClassHelper!!.getAllMethods(psiClass).size
         }
 
-        open fun fields(): Array<ScriptPsiFieldContext> {
+        override fun fields(): Array<ScriptPsiFieldContext> {
             return jvmClassHelper!!.getAllFields(psiClass)
                 .mapToTypedArray { ScriptPsiFieldContext(it) }
         }
@@ -772,6 +781,7 @@ abstract class ScriptRuleParser : RuleParser {
                     logger!!.info("try infer return type of method[" + PsiClassUtils.fullNameOfMethod(psiMethod) + "]")
                     methodReturnInferHelper!!.inferReturn(psiMethod)
                 }
+
                 else -> psiClassHelper!!.getTypeObject(
                     psiType, psiMethod,
                     JsonOption.NONE.or(readGetter, readSetter)
@@ -802,6 +812,7 @@ abstract class ScriptRuleParser : RuleParser {
                     logger!!.info("try infer return type of method[" + PsiClassUtils.fullNameOfMethod(psiMethod) + "]")
                     methodReturnInferHelper!!.inferReturn(psiMethod)
                 }
+
                 else -> psiClassHelper!!.getTypeObject(
                     duckType, psiMethod, JsonOption.NONE
                         .or(readGetter, readSetter)
@@ -900,11 +911,11 @@ abstract class ScriptRuleParser : RuleParser {
         }
 
         @ScriptIgnore
-        override fun getCore(): Any? {
+        override fun getCore(): Any {
             return explicitParam
         }
 
-        override fun method(): ScriptPsiMethodContext? {
+        override fun method(): ScriptPsiMethodContext {
             return ScriptExplicitMethodContext(explicitParam.containMethod())
         }
 
@@ -915,6 +926,10 @@ abstract class ScriptRuleParser : RuleParser {
 
     @ScriptTypeName("class")
     interface ScriptClassContext {
+
+        fun methods(): Array<ScriptPsiMethodContext>
+
+        fun fields(): Array<ScriptPsiFieldContext>
 
         fun isExtend(superClass: String): Boolean
 
@@ -1047,14 +1062,14 @@ abstract class ScriptRuleParser : RuleParser {
             }
         }
 
-        fun methods(): Array<ScriptPsiMethodContext> {
+        override fun methods(): Array<ScriptPsiMethodContext> {
             return (getResource() as? PsiClass)?.let { psiElement ->
                 return@let jvmClassHelper!!.getAllMethods(psiElement)
                     .mapToTypedArray { ScriptPsiMethodContext(it) }
             } ?: emptyArray()
         }
 
-        fun fields(): Array<ScriptPsiFieldContext> {
+        override fun fields(): Array<ScriptPsiFieldContext> {
             return (getResource() as? PsiClass)?.let { psiElement ->
                 return@let jvmClassHelper!!.getAllFields(psiElement)
                     .mapToTypedArray { ScriptPsiFieldContext(it) }
@@ -1269,14 +1284,14 @@ abstract class ScriptRuleParser : RuleParser {
             }
         }
 
-        fun methods(): Array<ScriptPsiMethodContext> {
+        override fun methods(): Array<ScriptPsiMethodContext> {
             return (getResource() as? PsiClass)?.let { psiElement ->
                 return@let jvmClassHelper!!.getAllMethods(psiElement)
                     .mapToTypedArray { ScriptPsiMethodContext(it) }
             } ?: emptyArray()
         }
 
-        fun fields(): Array<ScriptPsiFieldContext> {
+        override fun fields(): Array<ScriptPsiFieldContext> {
             return (getResource() as? PsiClass)?.let { psiElement ->
                 return@let jvmClassHelper!!.getAllFields(psiElement)
                     .mapToTypedArray { ScriptPsiFieldContext(it) }

--- a/idea-plugin/src/main/resources/.recommend.easy.api.config
+++ b/idea-plugin/src/main/resources/.recommend.easy.api.config
@@ -168,7 +168,7 @@ param.after=groovy:```
 param.required=@javax.validation.constraints.NotBlank
 param.required=@javax.validation.constraints.NotNull
 param.required=@javax.validation.constraints.NotEmpty
-check_groups=groovy:```
+check_groups=```
     for(annMap in annMaps){
         def fieldGroups = annMap["groups"] ?: [helper.findClass("javax.validation.groups.Default")]
         def paramGroups = session.get("json-group") ?: [helper.findClass("javax.validation.groups.Default")]

--- a/idea-plugin/src/test/kotlin/com/itangcent/idea/plugin/rule/ScriptClassContextBaseTest.kt
+++ b/idea-plugin/src/test/kotlin/com/itangcent/idea/plugin/rule/ScriptClassContextBaseTest.kt
@@ -30,6 +30,9 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
     internal lateinit var iResultPsiClass: PsiClass
     internal lateinit var userCtrlPsiClass: PsiClass
     internal lateinit var commentDemoPsiClass: PsiClass
+    internal lateinit var addPsiClass: PsiClass
+    internal lateinit var updatePsiClass: PsiClass
+    internal lateinit var validationGroupedDemoDtoPsiClass: PsiClass
 
     override fun beforeBind() {
         super.beforeBind()
@@ -54,6 +57,11 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
         loadFile("spring/RestController.java")
         userCtrlPsiClass = loadClass("api/UserCtrl.java")!!
         commentDemoPsiClass = loadClass("model/CommentDemo.java")!!
+        loadClass("validation/NotEmpty.java")
+        loadClass("validation/NotNull.java")
+        addPsiClass = loadClass("constant/Add.java")!!
+        updatePsiClass = loadClass("constant/Update.java")!!
+        validationGroupedDemoDtoPsiClass = loadClass("model/ValidationGroupedDemoDto.java")!!
     }
 
     override fun bind(builder: ActionContext.ActionContextBuilder) {
@@ -93,8 +101,10 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
     fun testHasAnn() {
         assertFalse(objectPsiClass.asScriptContext().hasAnn("org.springframework.web.bind.annotation.RestController"))
         assertFalse(integerPsiClass.asScriptContext().hasAnn("org.springframework.web.bind.annotation.RestController"))
-        assertFalse(collectionPsiClass.asScriptContext()
-            .hasAnn("org.springframework.web.bind.annotation.RestController"))
+        assertFalse(
+            collectionPsiClass.asScriptContext()
+                .hasAnn("org.springframework.web.bind.annotation.RestController")
+        )
         assertFalse(modelPsiClass.asScriptContext().hasAnn("org.springframework.web.bind.annotation.RestController"))
         assertTrue(userCtrlPsiClass.asScriptContext().hasAnn("org.springframework.web.bind.annotation.RestController"))
     }
@@ -104,8 +114,10 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
      */
     fun testAnn() {
         assertNull(objectPsiClass.asScriptContext().ann("org.springframework.web.bind.annotation.RequestMapping"))
-        assertEquals("user",
-            userCtrlPsiClass.asScriptContext().ann("org.springframework.web.bind.annotation.RequestMapping"))
+        assertEquals(
+            "user",
+            userCtrlPsiClass.asScriptContext().ann("org.springframework.web.bind.annotation.RequestMapping")
+        )
     }
 
     /**
@@ -113,8 +125,20 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
      */
     fun testAnnMap() {
         assertNull(objectPsiClass.asScriptContext().annMap("org.springframework.web.bind.annotation.RequestMapping"))
-        assertEquals(mapOf("value" to "user"),
-            userCtrlPsiClass.asScriptContext().annMap("org.springframework.web.bind.annotation.RequestMapping"))
+        assertEquals(
+            mapOf("value" to "user"),
+            userCtrlPsiClass.asScriptContext().annMap("org.springframework.web.bind.annotation.RequestMapping")
+        )
+        assertEquals(
+            mapOf("groups" to listOf(addPsiClass.asScriptContext())),
+            validationGroupedDemoDtoPsiClass.asClassContext()
+                .fields()[0].annMap("javax.validation.constraints.NotNull")
+        )
+        assertEquals(
+            mapOf("groups" to updatePsiClass.asScriptContext()),
+            validationGroupedDemoDtoPsiClass.asClassContext()
+                .fields()[1].annMap("javax.validation.constraints.NotEmpty")
+        )
     }
 
     /**
@@ -122,8 +146,10 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
      */
     fun testAnnMaps() {
         assertNull(objectPsiClass.asScriptContext().annMap("org.springframework.web.bind.annotation.RequestMapping"))
-        assertEquals(listOf(mapOf("value" to "user")),
-            userCtrlPsiClass.asScriptContext().annMaps("org.springframework.web.bind.annotation.RequestMapping"))
+        assertEquals(
+            listOf(mapOf("value" to "user")),
+            userCtrlPsiClass.asScriptContext().annMaps("org.springframework.web.bind.annotation.RequestMapping")
+        )
     }
 
     /**
@@ -131,8 +157,10 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
      */
     fun testAnnValue() {
         assertNull(objectPsiClass.asScriptContext().annValue("org.springframework.web.bind.annotation.RequestMapping"))
-        assertEquals("user",
-            userCtrlPsiClass.asScriptContext().annValue("org.springframework.web.bind.annotation.RequestMapping"))
+        assertEquals(
+            "user",
+            userCtrlPsiClass.asScriptContext().annValue("org.springframework.web.bind.annotation.RequestMapping")
+        )
     }
 
     /**
@@ -172,145 +200,149 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
     }
 
     fun testSourceCode() {
-        assertEquals("class Model {\n" +
-                "\n" +
-                "    /**\n" +
-                "     * string field\n" +
-                "     */\n" +
-                "    @JsonProperty(\"s\")\n" +
-                "    private String str;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * integer field\n" +
-                "     */\n" +
-                "    private Integer integer;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * stringList field\n" +
-                "     */\n" +
-                "    private List<String> stringList;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * integerArray field\n" +
-                "     */\n" +
-                "    private Integer[] integerArray;\n" +
-                "\n" +
-                "    public String getStr() {\n" +
-                "        return str;\n" +
-                "    }\n" +
-                "\n" +
-                "    public void setStr(String str) {\n" +
-                "        this.str = str;\n" +
-                "    }\n" +
-                "\n" +
-                "    public Integer getInteger() {\n" +
-                "        return integer;\n" +
-                "    }\n" +
-                "\n" +
-                "    public void setInteger(Integer integer) {\n" +
-                "        this.integer = integer;\n" +
-                "    }\n" +
-                "\n" +
-                "    public List<String> getStringList() {\n" +
-                "        return stringList;\n" +
-                "    }\n" +
-                "\n" +
-                "    public void setStringList(List<String> stringList) {\n" +
-                "        this.stringList = stringList;\n" +
-                "    }\n" +
-                "\n" +
-                "    public Integer[] getIntegerArray() {\n" +
-                "        return integerArray;\n" +
-                "    }\n" +
-                "\n" +
-                "    public void setIntegerArray(Integer[] integerArray) {\n" +
-                "        this.integerArray = integerArray;\n" +
-                "    }\n" +
-                "\n" +
-                "    public void setOnlySet(String onlySet) {\n" +
-                "\n" +
-                "    }\n" +
-                "\n" +
-                "    public String getOnlyGet() {\n" +
-                "\n" +
-                "    }\n" +
-                "}", modelPsiClass.asScriptContext().sourceCode())
-        assertEquals("/**\n" +
-                " * A demo class for test comments\n" +
-                " *\n" +
-                " * @author tangcent\n" +
-                " */\n" +
-                "public class CommentDemo {\n" +
-                "\n" +
-                "    /**\n" +
-                "     * single line\n" +
-                "     *\n" +
-                "     * @single\n" +
-                "     * @desc low case of A\n" +
-                "     */\n" +
-                "    private String a;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * multi-line\n" +
-                "     * second line\n" +
-                "     *\n" +
-                "     * @multi\n" +
-                "     * @module x\n" +
-                "     * x1 x2 x3\n" +
-                "     * @module y\n" +
-                "     */\n" +
-                "    private String b;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * head line\n" +
-                "     * second line\n" +
-                "     * <pre>\n" +
-                "     *     {\n" +
-                "     *         \"a\":\"b\",\n" +
-                "     *         \"c\":{\n" +
-                "     *              \"x\":[\"y\"]\n" +
-                "     *         }\n" +
-                "     *     }\n" +
-                "     * </pre>\n" +
-                "     * see @{link somelink}\n" +
-                "     * tail line\n" +
-                "     */\n" +
-                "    private String c;\n" +
-                "\n" +
-                "    /**\n" +
-                "     * head line\n" +
-                "     * second line\n" +
-                "     * <pre>\n" +
-                "     *\n" +
-                "     *     {\n" +
-                "     *         \"a\":\"b\",\n" +
-                "     *         \"c\":{\n" +
-                "     *              \"x\":[\"y\"]\n" +
-                "     *         }\n" +
-                "     *     }\n" +
-                "     *\n" +
-                "     * </pre>\n" +
-                "     * <p>\n" +
-                "     * see @{link somelink}\n" +
-                "     * tail line\n" +
-                "     */\n" +
-                "    private String d;\n" +
-                "\n" +
-                "    private String e;//E is a mathematical constant approximately equal to 2.71828\n" +
-                "\n" +
-                "    /**\n" +
-                "     * R, or r, is the eighteenth letter of the modern English alphabet and the ISO basic Latin alphabet.\n" +
-                "     */\n" +
-                "    private String r;//It's before s\n" +
-                "\n" +
-                "    /**\n" +
-                "     * @param a A\n" +
-                "     * @param b B\n" +
-                "     */\n" +
-                "    private String methodA(int a, int b) {\n" +
-                "\n" +
-                "    }\n" +
-                "}", commentDemoPsiClass.asScriptContext().sourceCode())
+        assertEquals(
+            "class Model {\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * string field\n" +
+                    "     */\n" +
+                    "    @JsonProperty(\"s\")\n" +
+                    "    private String str;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * integer field\n" +
+                    "     */\n" +
+                    "    private Integer integer;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * stringList field\n" +
+                    "     */\n" +
+                    "    private List<String> stringList;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * integerArray field\n" +
+                    "     */\n" +
+                    "    private Integer[] integerArray;\n" +
+                    "\n" +
+                    "    public String getStr() {\n" +
+                    "        return str;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public void setStr(String str) {\n" +
+                    "        this.str = str;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public Integer getInteger() {\n" +
+                    "        return integer;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public void setInteger(Integer integer) {\n" +
+                    "        this.integer = integer;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public List<String> getStringList() {\n" +
+                    "        return stringList;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public void setStringList(List<String> stringList) {\n" +
+                    "        this.stringList = stringList;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public Integer[] getIntegerArray() {\n" +
+                    "        return integerArray;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public void setIntegerArray(Integer[] integerArray) {\n" +
+                    "        this.integerArray = integerArray;\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public void setOnlySet(String onlySet) {\n" +
+                    "\n" +
+                    "    }\n" +
+                    "\n" +
+                    "    public String getOnlyGet() {\n" +
+                    "\n" +
+                    "    }\n" +
+                    "}", modelPsiClass.asScriptContext().sourceCode()
+        )
+        assertEquals(
+            "/**\n" +
+                    " * A demo class for test comments\n" +
+                    " *\n" +
+                    " * @author tangcent\n" +
+                    " */\n" +
+                    "public class CommentDemo {\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * single line\n" +
+                    "     *\n" +
+                    "     * @single\n" +
+                    "     * @desc low case of A\n" +
+                    "     */\n" +
+                    "    private String a;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * multi-line\n" +
+                    "     * second line\n" +
+                    "     *\n" +
+                    "     * @multi\n" +
+                    "     * @module x\n" +
+                    "     * x1 x2 x3\n" +
+                    "     * @module y\n" +
+                    "     */\n" +
+                    "    private String b;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * head line\n" +
+                    "     * second line\n" +
+                    "     * <pre>\n" +
+                    "     *     {\n" +
+                    "     *         \"a\":\"b\",\n" +
+                    "     *         \"c\":{\n" +
+                    "     *              \"x\":[\"y\"]\n" +
+                    "     *         }\n" +
+                    "     *     }\n" +
+                    "     * </pre>\n" +
+                    "     * see @{link somelink}\n" +
+                    "     * tail line\n" +
+                    "     */\n" +
+                    "    private String c;\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * head line\n" +
+                    "     * second line\n" +
+                    "     * <pre>\n" +
+                    "     *\n" +
+                    "     *     {\n" +
+                    "     *         \"a\":\"b\",\n" +
+                    "     *         \"c\":{\n" +
+                    "     *              \"x\":[\"y\"]\n" +
+                    "     *         }\n" +
+                    "     *     }\n" +
+                    "     *\n" +
+                    "     * </pre>\n" +
+                    "     * <p>\n" +
+                    "     * see @{link somelink}\n" +
+                    "     * tail line\n" +
+                    "     */\n" +
+                    "    private String d;\n" +
+                    "\n" +
+                    "    private String e;//E is a mathematical constant approximately equal to 2.71828\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * R, or r, is the eighteenth letter of the modern English alphabet and the ISO basic Latin alphabet.\n" +
+                    "     */\n" +
+                    "    private String r;//It's before s\n" +
+                    "\n" +
+                    "    /**\n" +
+                    "     * @param a A\n" +
+                    "     * @param b B\n" +
+                    "     */\n" +
+                    "    private String methodA(int a, int b) {\n" +
+                    "\n" +
+                    "    }\n" +
+                    "}", commentDemoPsiClass.asScriptContext().sourceCode()
+        )
     }
 
     fun testDefineCode() {
@@ -432,106 +464,150 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
     fun testToJson() {
         assertEquals("{}", objectPsiClass.asClassContext().toJson(true, true))
 
-        assertEquals("{\n" +
-                "  \"str\": \"\",\n" +
-                "  \"integer\": 0,\n" +
-                "  \"stringList\": [\n" +
-                "    \"\"\n" +
-                "  ],\n" +
-                "  \"integerArray\": [\n" +
-                "    0\n" +
-                "  ],\n" +
-                "  \"onlySet\": \"\",\n" +
-                "  \"onlyGet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson(true, true))
-        assertEquals("{\n" +
-                "  \"str\": \"\",\n" +
-                "  \"integer\": 0,\n" +
-                "  \"stringList\": [\n" +
-                "    \"\"\n" +
-                "  ],\n" +
-                "  \"integerArray\": [\n" +
-                "    0\n" +
-                "  ],\n" +
-                "  \"onlyGet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson(true, false))
-        assertEquals("{\n" +
-                "  \"str\": \"\",\n" +
-                "  \"integer\": 0,\n" +
-                "  \"stringList\": [\n" +
-                "    \"\"\n" +
-                "  ],\n" +
-                "  \"integerArray\": [\n" +
-                "    0\n" +
-                "  ],\n" +
-                "  \"onlySet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson(false, true))
-        assertEquals("{\n" +
-                "  \"str\": \"\",\n" +
-                "  \"integer\": 0,\n" +
-                "  \"stringList\": [\n" +
-                "    \"\"\n" +
-                "  ],\n" +
-                "  \"integerArray\": [\n" +
-                "    0\n" +
-                "  ]\n" +
-                "}", modelPsiClass.asClassContext().toJson(false, false))
+        assertEquals(
+            "{\n" +
+                    "  \"str\": \"\",\n" +
+                    "  \"integer\": 0,\n" +
+                    "  \"stringList\": [\n" +
+                    "    \"\"\n" +
+                    "  ],\n" +
+                    "  \"integerArray\": [\n" +
+                    "    0\n" +
+                    "  ],\n" +
+                    "  \"onlySet\": \"\",\n" +
+                    "  \"onlyGet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"str\": \"\",\n" +
+                    "  \"integer\": 0,\n" +
+                    "  \"stringList\": [\n" +
+                    "    \"\"\n" +
+                    "  ],\n" +
+                    "  \"integerArray\": [\n" +
+                    "    0\n" +
+                    "  ],\n" +
+                    "  \"onlyGet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"str\": \"\",\n" +
+                    "  \"integer\": 0,\n" +
+                    "  \"stringList\": [\n" +
+                    "    \"\"\n" +
+                    "  ],\n" +
+                    "  \"integerArray\": [\n" +
+                    "    0\n" +
+                    "  ],\n" +
+                    "  \"onlySet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"str\": \"\",\n" +
+                    "  \"integer\": 0,\n" +
+                    "  \"stringList\": [\n" +
+                    "    \"\"\n" +
+                    "  ],\n" +
+                    "  \"integerArray\": [\n" +
+                    "    0\n" +
+                    "  ]\n" +
+                    "}", modelPsiClass.asClassContext().toJson(false, false)
+        )
 
-        assertEquals("{\n" +
-                "  \"\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson(true, true))
-        assertEquals("{\n" +
-                "  \"\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson(true, false))
-        assertEquals("{\n" +
-                "  \"\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson(false, true))
-        assertEquals("{\n" +
-                "  \"\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson(false, false))
+        assertEquals(
+            "{\n" +
+                    "  \"\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson(false, false)
+        )
 
-        assertTrue(arrayOf("[]", "[\n" +
-                "  {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson(true, true)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "  {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson(true, false)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "  {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson(false, true)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "  {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson(false, false)))
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "  {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson(true, true))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "  {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson(true, false))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "  {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson(false, true))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "  {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson(false, false))
+        )
 
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\",\n" +
-                "  \"data\": {}\n" +
-                "}", resultPsiClass.asClassContext().toJson(true, true))
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\",\n" +
-                "  \"data\": {}\n" +
-                "}", resultPsiClass.asClassContext().toJson(true, false))
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\",\n" +
-                "  \"data\": {}\n" +
-                "}", resultPsiClass.asClassContext().toJson(false, true))
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\",\n" +
-                "  \"data\": {}\n" +
-                "}", resultPsiClass.asClassContext().toJson(false, false))
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\",\n" +
+                    "  \"data\": {}\n" +
+                    "}", resultPsiClass.asClassContext().toJson(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\",\n" +
+                    "  \"data\": {}\n" +
+                    "}", resultPsiClass.asClassContext().toJson(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\",\n" +
+                    "  \"data\": {}\n" +
+                    "}", resultPsiClass.asClassContext().toJson(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\",\n" +
+                    "  \"data\": {}\n" +
+                    "}", resultPsiClass.asClassContext().toJson(false, false)
+        )
 
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\"\n" +
-                "}", iResultPsiClass.asClassContext().toJson(true, true))
-        assertEquals("{\n" +
-                "  \"code\": 0,\n" +
-                "  \"msg\": \"\"\n" +
-                "}", iResultPsiClass.asClassContext().toJson(true, false))
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\"\n" +
+                    "}", iResultPsiClass.asClassContext().toJson(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "  \"code\": 0,\n" +
+                    "  \"msg\": \"\"\n" +
+                    "}", iResultPsiClass.asClassContext().toJson(true, false)
+        )
         assertEquals("{}", iResultPsiClass.asClassContext().toJson(false, true))
         assertEquals("{}", iResultPsiClass.asClassContext().toJson(false, false))
     }
@@ -539,106 +615,150 @@ abstract class ScriptClassContextBaseTest : PluginContextLightCodeInsightFixture
     fun testToJson5() {
         assertEquals("{}", objectPsiClass.asClassContext().toJson5(true, true))
 
-        assertEquals("{\n" +
-                "    \"str\": \"\", //string field\n" +
-                "    \"integer\": 0, //integer field\n" +
-                "    \"stringList\": [ //stringList field\n" +
-                "        \"\"\n" +
-                "    ],\n" +
-                "    \"integerArray\": [ //integerArray field\n" +
-                "        0\n" +
-                "    ],\n" +
-                "    \"onlySet\": \"\",\n" +
-                "    \"onlyGet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson5(true, true))
-        assertEquals("{\n" +
-                "    \"str\": \"\", //string field\n" +
-                "    \"integer\": 0, //integer field\n" +
-                "    \"stringList\": [ //stringList field\n" +
-                "        \"\"\n" +
-                "    ],\n" +
-                "    \"integerArray\": [ //integerArray field\n" +
-                "        0\n" +
-                "    ],\n" +
-                "    \"onlyGet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson5(true, false))
-        assertEquals("{\n" +
-                "    \"str\": \"\", //string field\n" +
-                "    \"integer\": 0, //integer field\n" +
-                "    \"stringList\": [ //stringList field\n" +
-                "        \"\"\n" +
-                "    ],\n" +
-                "    \"integerArray\": [ //integerArray field\n" +
-                "        0\n" +
-                "    ],\n" +
-                "    \"onlySet\": \"\"\n" +
-                "}", modelPsiClass.asClassContext().toJson5(false, true))
-        assertEquals("{\n" +
-                "    \"str\": \"\", //string field\n" +
-                "    \"integer\": 0, //integer field\n" +
-                "    \"stringList\": [ //stringList field\n" +
-                "        \"\"\n" +
-                "    ],\n" +
-                "    \"integerArray\": [ //integerArray field\n" +
-                "        0\n" +
-                "    ]\n" +
-                "}", modelPsiClass.asClassContext().toJson5(false, false))
+        assertEquals(
+            "{\n" +
+                    "    \"str\": \"\", //string field\n" +
+                    "    \"integer\": 0, //integer field\n" +
+                    "    \"stringList\": [ //stringList field\n" +
+                    "        \"\"\n" +
+                    "    ],\n" +
+                    "    \"integerArray\": [ //integerArray field\n" +
+                    "        0\n" +
+                    "    ],\n" +
+                    "    \"onlySet\": \"\",\n" +
+                    "    \"onlyGet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson5(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"str\": \"\", //string field\n" +
+                    "    \"integer\": 0, //integer field\n" +
+                    "    \"stringList\": [ //stringList field\n" +
+                    "        \"\"\n" +
+                    "    ],\n" +
+                    "    \"integerArray\": [ //integerArray field\n" +
+                    "        0\n" +
+                    "    ],\n" +
+                    "    \"onlyGet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson5(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"str\": \"\", //string field\n" +
+                    "    \"integer\": 0, //integer field\n" +
+                    "    \"stringList\": [ //stringList field\n" +
+                    "        \"\"\n" +
+                    "    ],\n" +
+                    "    \"integerArray\": [ //integerArray field\n" +
+                    "        0\n" +
+                    "    ],\n" +
+                    "    \"onlySet\": \"\"\n" +
+                    "}", modelPsiClass.asClassContext().toJson5(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"str\": \"\", //string field\n" +
+                    "    \"integer\": 0, //integer field\n" +
+                    "    \"stringList\": [ //stringList field\n" +
+                    "        \"\"\n" +
+                    "    ],\n" +
+                    "    \"integerArray\": [ //integerArray field\n" +
+                    "        0\n" +
+                    "    ]\n" +
+                    "}", modelPsiClass.asClassContext().toJson5(false, false)
+        )
 
-        assertEquals("{\n" +
-                "    \"key\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson5(true, true))
-        assertEquals("{\n" +
-                "    \"key\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson5(true, false))
-        assertEquals("{\n" +
-                "    \"key\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson5(false, true))
-        assertEquals("{\n" +
-                "    \"key\": null\n" +
-                "}", mapPsiClass.asClassContext().toJson5(false, false))
+        assertEquals(
+            "{\n" +
+                    "    \"key\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson5(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"key\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson5(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"key\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson5(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"key\": null\n" +
+                    "}", mapPsiClass.asClassContext().toJson5(false, false)
+        )
 
-        assertTrue(arrayOf("[]", "[\n" +
-                "    {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson5(true, true)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "    {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson5(true, false)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "    {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson5(false, true)))
-        assertTrue(arrayOf("[]", "[\n" +
-                "    {}\n" +
-                "]").contains(collectionPsiClass.asClassContext().toJson5(false, false)))
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "    {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson5(true, true))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "    {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson5(true, false))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "    {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson5(false, true))
+        )
+        assertTrue(
+            arrayOf(
+                "[]", "[\n" +
+                        "    {}\n" +
+                        "]"
+            ).contains(collectionPsiClass.asClassContext().toJson5(false, false))
+        )
 
-        assertEquals("{\n" +
-                "    \"code\": 0, //response code\n" +
-                "    \"msg\": \"\", //message\n" +
-                "    \"data\": {} //response data\n" +
-                "}", resultPsiClass.asClassContext().toJson5(true, true))
-        assertEquals("{\n" +
-                "    \"code\": 0, //response code\n" +
-                "    \"msg\": \"\", //message\n" +
-                "    \"data\": {} //response data\n" +
-                "}", resultPsiClass.asClassContext().toJson5(true, false))
-        assertEquals("{\n" +
-                "    \"code\": 0, //response code\n" +
-                "    \"msg\": \"\", //message\n" +
-                "    \"data\": {} //response data\n" +
-                "}", resultPsiClass.asClassContext().toJson5(false, true))
-        assertEquals("{\n" +
-                "    \"code\": 0, //response code\n" +
-                "    \"msg\": \"\", //message\n" +
-                "    \"data\": {} //response data\n" +
-                "}", resultPsiClass.asClassContext().toJson5(false, false))
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0, //response code\n" +
+                    "    \"msg\": \"\", //message\n" +
+                    "    \"data\": {} //response data\n" +
+                    "}", resultPsiClass.asClassContext().toJson5(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0, //response code\n" +
+                    "    \"msg\": \"\", //message\n" +
+                    "    \"data\": {} //response data\n" +
+                    "}", resultPsiClass.asClassContext().toJson5(true, false)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0, //response code\n" +
+                    "    \"msg\": \"\", //message\n" +
+                    "    \"data\": {} //response data\n" +
+                    "}", resultPsiClass.asClassContext().toJson5(false, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0, //response code\n" +
+                    "    \"msg\": \"\", //message\n" +
+                    "    \"data\": {} //response data\n" +
+                    "}", resultPsiClass.asClassContext().toJson5(false, false)
+        )
 
-        assertEquals("{\n" +
-                "    \"code\": 0,\n" +
-                "    \"msg\": \"\"\n" +
-                "}", iResultPsiClass.asClassContext().toJson5(true, true))
-        assertEquals("{\n" +
-                "    \"code\": 0,\n" +
-                "    \"msg\": \"\"\n" +
-                "}", iResultPsiClass.asClassContext().toJson5(true, false))
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0,\n" +
+                    "    \"msg\": \"\"\n" +
+                    "}", iResultPsiClass.asClassContext().toJson5(true, true)
+        )
+        assertEquals(
+            "{\n" +
+                    "    \"code\": 0,\n" +
+                    "    \"msg\": \"\"\n" +
+                    "}", iResultPsiClass.asClassContext().toJson5(true, false)
+        )
         assertEquals("{}", iResultPsiClass.asClassContext().toJson5(false, true))
         assertEquals("{}", iResultPsiClass.asClassContext().toJson5(false, false))
     }

--- a/idea-plugin/src/test/resources/model/ValidationGroupedDemoDto.java
+++ b/idea-plugin/src/test/resources/model/ValidationGroupedDemoDto.java
@@ -8,7 +8,7 @@ import javax.validation.constraints.NotNull;
 
 public class ValidationGroupedDemoDto {
 
-    @NotNull(groups = Add.class)
+    @NotNull(groups = {Add.class})
     private String strForAdd;
 
     @NotEmpty(groups = Update.class)

